parachains/runtimes/assets/statemine/src/constants.rs

```diff
@@ -55,13 +55,45 @@ pub mod fee {
 	///   - Setting it to `0` will essentially disable the weight fee.
 	///   - Setting it to `1` will cause the literal `#[weight = x]` values to be charged.
 	pub struct WeightToFee;
-	impl WeightToFeePolynomial for WeightToFee {
+	impl frame_support::weights::WeightToFee for WeightToFee {
+		type Balance = Balance;
+
+		fn weight_to_fee(weight: &Weight) -> Self::Balance {
+			let time_poly: FeePolynomial<Balance> = RefTimeToFee::polynomial().into();
+			let proof_poly: FeePolynomial<Balance> = ProofSizeToFee::polynomial().into();
+
+			time_poly.eval(weight.ref_time()).max(proof_poly.eval(weight.proof_size()))
```

Comment:
Why it is a max not add? It makes some senses but we should have this reason written down somewhere.